server/tests/stt_ntb_ninjs_parse_test.py

```diff
@@ -0,0 +1,226 @@
+                    {"is_active": "true", "qcode": "14", "name": "Ulkomaat"},
+                    {"is_active": "true", "name": "Business wire", "qcode": "1"},
+                    {"is_active": "true", "name": "Holvi", "qcode": "23"},
+                    {
+                        "is_active": "true",
+                        "name": "Julkishallinnon tiedotepalvelu",
+                        "qcode": "2",
+                    },
+                    {"is_active": "true", "name": "Kotimaa", "qcode": "3"},
+                    {"is_active": "true", "name": "Kulttuuri", "qcode": "4"},
+                    {"is_active": "true", "name": "Merkkipäiväpalvelu", "qcode": "5"},
+                    {"is_active": "true", "name": "Muuta", "qcode": "6"},
+                    {"is_active": "true", "name": "Peliuutiset", "qcode": "8"},
+                    {"is_active": "true", "name": "Politiikka", "qcode": "9"},
+                    {"is_active": "true", "name": "Päivälista", "qcode": "21"},
+                    {"is_active": "true", "name": "Talous", "qcode": "11"},
+                    {"is_active": "true", "name": "Tiedotepalvelu", "qcode": "12"},
+                    {
+                        "is_active": "true",
+                        "name": "Toimituksille tiedoksi",
+                        "qcode": "13",
+                    },
+                    {"is_active": "true", "name": "Urheilu", "qcode": "16"},
+                    {"is_active": "true", "name": "Uutiskooste", "qcode": "22"},
+                    {"is_active": "true", "name": "Viikon tärpit", "qcode": "19"},
+                    {"is_active": "true", "name": "Sähkeuutiset", "qcode": "10"},
+                ]
+            elif vocab_id == "stt_media_topics":
+                # Mock some media topics to ensure media topic parsing works
+                return [
+                    {"name": "Test Topic", "qcode": "20000023"},
+                    {"name": "Another Topic", "qcode": "20000021"},
+                ]
+            return []
+
+        mock_load_cv.side_effect = mock_cv_lookup
+
+        # Parse the item with mocked vocabulary to test proper category mapping
+        dirname = os.path.dirname(os.path.realpath(__file__))
+        fixture = os.path.join(dirname, "fixtures", self.fixture)
+        with self.ctx:
+            parser = STTTTNINJSParseFeedParser()
+            parsed_items = parser.parse(fixture)
+            item = parsed_items[0] if parsed_items else {}
+
+        # Test category mapping works correctly
+        anpa_qcodes = {
+            a.get("qcode") for a in item.get("anpa_category", []) if a.get("qcode")
+        }
+
+        # Ensure mapped category "14" is present (others may also be present, e.g., fallback)
+        assert "14" in anpa_qcodes
+
+        # Verify the category entry structure for qcode 14
+        anpa_category = item.get("anpa_category", [])
+        q14 = next((a for a in anpa_category if a.get("qcode") == "14"), None)
```

Comment:
would be easier to read if this whole `anpa_category` testing would be like:
```py
assert item["anpa_category"] == [{"name": "Ulkomaat", "qcode": "14"}]
```
instead of lines 196 to 209